manticore/core/smtlib/solver.py

```diff
@@ -276,7 +276,7 @@ def __del__(self):
         except Exception as e:
             logger.error(str(e))
 
-    def _reset(self, constraints=None):
+    def _reset(self, constraints: Optional[str] = None) -> None:
```

Comment:
Why is this not
```suggestion
    def _reset(self, constraints: Optional[Union[str, ConstraintSet]] = None) -> None:
```
like `_send`?

I see that you call `.to_string()` on the `constraints` to send to `_reset`. If it's the case that `_send` should always receive a `str`, then I think that type annotation should be changed/checked as well, at least for consistency. From my cursory search, it seems we only send `str`s to `_send`.

I'm generally in favor of more restricted types (i.e., not `Union`), if possible.

Reply:
So, you're asking why `_reset` doesn't have the same signature as `_send`?  Good question!

The change here _adds_ a type hint to `_reset`, giving it a tighter signature than what `_send` currently has.  The other half of this — tightening the type of `_send` to `def _send(self, cmd: str)` — ended up in my other open pull request (#1667).  Let me add that to _this_ pull request instead, for coherence.

I was trying to keep _this_ pull request focused on fixing type hints that are manifestly wrong at runtime (actually shown dynamically to not be accurate type hints).

Reply:
> I was trying to keep this pull request focused on fixing type hints that are manifestly wrong at runtime (actually shown dynamically to not be accurate type hints).

Ah. Okay, that makes sense. There's like 3 open PRs for type hints, and I couldn't keep the differences straight in my head haha